### PR TITLE
feat: add `checkov` spec

### PR DESCRIPTION
**Pull request by eurogig**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Added autocompletion spec for the open source IaC scanning tool Checkov by Bridgecrew.io

**What is the current behavior? (You can also link to an open issue here)**
No previous support for this CLI tool.

**What is the new behavior (if this is a feature change)?**
Support for Checkov!

**Additional info:**
Fig rocks!

### CODE REVIEW

**Comment by eurogig** [comment]
Sorry I didn't keep the to "feat: add checkov completion spec".  Got excited and missed that contributor detail.  Double checked all the args and tested the commands.

File: src/checkov.ts [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  name: "checkov",
+  description: "",
+  options: [
+    {
+      name: ["--help", "-h"],
+      description: "Show help for checkov",
+    },
+    {
+      name: ["--directory", "-d"],
+      description: "Show the version of checkov",
```
Comment (fedeci) [inline src/checkov.ts:23]: I think this is not the correct description

File: src/checkov.ts [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        template: "folders",
+      },
+    },
+    {
+      name: "--add-check",
+      description: "Generate a new check via CLI prompt",
+    },
+    {
+      name: ["--file", "-f"],
+      description: "IaC file(can not be used together with --directory)",
```
Comment (fedeci) [inline src/checkov.ts:35]:
```suggestion
      description: "IaC file",
      exclusiveOn: ["--file"],
```

File: src/checkov.ts [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+      args: {
+        name: "FILE",
+        template: "filepaths",
+      },
+    },
+    {
+      name: "--skip-path",
+      description:
+        "Path (file or directory) to skip, using regular expression logic, relative to current working directory. Word boundaries are not implicit; i.e., specifying 'dir1' will skip any directory or subdirectory named 'dir1'. Ignored with -f. Can be specified multiple times",
```
Comment (fedeci) [inline src/checkov.ts:44]: This seems to be variadic

File: src/checkov.ts [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+    }
+    return output.split("\n").map((branch) => {
+      return { name: branch.replace("*", "").trim(), description: "Branch" };
+    });
+  },
+};
+
+const completionSpec: Fig.Spec = {
+  name: "checkov",
+  description: "",
```
Comment (fedeci) [inline src/checkov.ts:15]: Could you add a short description for the tool please?

File: src/checkov.ts [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+    {
+      name: "--skip-check",
+      description:
+        "Filter scan to run all check except a specific check identifier (denylist). You can specify multiple checks separated by comma delimiter. You may also use the environment variable: CKV_SKIP_CHECK",
+      args: {
+        name: "CHECKS",
+      },
+    },
+    {
+      name: "----run-all-external-checks",
```
Comment (fedeci) [inline src/checkov.ts:67]: these are a lot of dashes 😄

File: src/checkov.ts [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+      description:
+        "Bridgecrew API key. You may also use the environment variable: BC_API_KEY",
+      args: {
+        name: "BC_API_KEY",
+      },
+    },
+    {
+      name: "--docker-image",
+      description:
+        "Scan docker images by name or ID. Only works with --bc-api-key flag",
```
Comment (fedeci) [inline src/checkov.ts:82]: This needs [`dependsOn`](https://fig.io/docs/reference/option#dependson)

File: src/checkov.ts [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+      description:
+        "Identity string of the repository, with form <repo_owner>/<repo_name>",
+      args: {
+        name: "REPO_ID",
+      },
+    },
+    {
+      name: ["--branch", "-b"],
+      description:
+        "Selected branch of the persisted repository. Only has effect when using the --bc-api-key flag",
```
Comment (fedeci) [inline src/checkov.ts:105]: Also this needs `dependsOn`

File: src/checkov.ts [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+    },
+    {
+      name: "--show-config",
+      description:
+        "Prints all arguments and config settings and where they came from (eg. commandline, config file, environment variable or default)",
+    },
+    {
+      name: "--create-baseline",
+      description:
+        "Save all current results to a '.checkov.baseline' file so future runs will only flag new findings. Works only with `--directory` flag",
```
Comment (fedeci) [inline src/checkov.ts:196]: this probably is missing `dependsOn`

File: src/checkov.ts [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+      description:
+        "Custom CA (bundle) fila. You may also use the environment variablee: CA_CERTIFICATE",
+      args: {
+        name: "CA_CERTIFICATE",
+      },
+    },
+    {
+      name: "--repo-root-for-plan-enrichment",
+      description:
+        "Directory containing the hcl code used to generate a given plan file. Use with -f FILE",
```
Comment (fedeci) [inline src/checkov.ts:168]: This is missing dependsOn

File: src/checkov.ts [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+        name: "DOWNLOAD_EXTERNAL_MODULES",
+      },
+    },
+    {
+      name: "--var-file",
+      description:
+        "Variable files to load in addition to the default files (see https://www.terraform.io/docs/language/values/variables.html#variable-definitions-tfvars-files). Currently only supported for source Terraform (.tf file), and Helm chart scans. Requires using --directory, NOT --file",
+      args: {
+        name: "VAR_FILE",
+      },
```
Comment (fedeci) [inline src/checkov.ts:140]: This is both depending on `--directory` and exclusive on `--file`. A the same `--file` should be exclusive on this one.

File: src/checkov.ts [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+      args: {
+        name: "BRANCH",
+        generators: branches,
+      },
+    },
+    {
+      name: "--skip-fixes",
+      description:
+        "Do not download fixed resource templates from Bridgecrew. Only has an effect when using the --bc-api-key flag",
+    },
```
Comment (fedeci) [inline src/checkov.ts:115]: This `dependsOn` bc-api-key

File: src/checkov.ts [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+    {
+      name: "--skip-suppressions",
+      description:
+        "Do not download preconfigured suppressions from the Bridgecrew platform. Code comment suppressions will still be honored. Only has an effect when using the --bc-api-key flag",
+    },
+    {
+      name: "--skip-policy-download",
```
Comment (fedeci) [inline src/checkov.ts:122]: This dependsOn bc-api-key

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution! I left some suggestions to improve the spec but it looks great🎉

File: src/checkov.ts (hunk 3 above)
Comment (eurogig) [reply src/checkov.ts:44]: ...and you would be correct.  Feel free to open a pull request on our tool to fix it.  This exercise of providing support for Checkov has been interesting in me having the same revelations.  Unfortunately that description is cut and paste from our help.

File: src/checkov.ts (hunk 4 above)
Comment (eurogig) [reply src/checkov.ts:15]: Can do.  Saw a lot of files in the src directory without descriptions so wasn't sure how key this was.  Will add one.

File: src/checkov.ts (hunk 5 above)
Comment (eurogig) [reply src/checkov.ts:67]: Sh****t.. Will fix.

File: src/checkov.ts (hunk 6 above)
Comment (eurogig) [reply src/checkov.ts:82]: Didn't see that feature.  Like it.  Will take all these on and resubmit.

File: src/checkov.ts (hunk 4 above)
Comment (fedeci) [reply src/checkov.ts:15]: Thanks!

File: src/checkov.ts (hunk 3 above)
Comment (fedeci) [reply src/checkov.ts:44]: I meant we can add the variadic prop to the argument of the option in the spec😄

File: src/checkov.ts (hunk 6 above)
Comment (fedeci) [reply src/checkov.ts:82]: Great!

File: src/checkov.ts (hunk 3 above)
Comment (eurogig) [reply src/checkov.ts:44]: Oh oh... yeah that.  Ok thanks!

File: src/checkov.ts (hunk 2 above)
Comment (eurogig) [reply src/checkov.ts:37]: It was exclusive on --directory actually.  Added.

File: src/checkov.ts (hunk 3 above)
Comment (eurogig) [reply src/checkov.ts:47]: Changes made

File: src/checkov.ts (hunk 8 above)
Comment (eurogig) [reply src/checkov.ts:210]: Ok should be done.  All suggestions should be resolved.

**Comment by eurogig** [comment]
@fedeci thanks!  All changes should be resolved now.

**Review by fedeci — APPROVED** [review]
Thank you!

**Pull request merged**